### DISCUSSION TITLE
How to dynamically load pre-rendered pages?

### DISCUSSION THREAD

**Discussion by AlexMili** [nuxt/nuxt#28442] [Questions] [answered]
I pre-rendered my whole website but the problem is that I have a "blank screen" between each pages. What I would like is keeping the header and the footer and/or have a loading screen between each route.

`NuxtLoadingIndicator` doesn't seem to work in prerender.

I don't mind having the header and footer not pre-rendered beacause they don't use much but I would like to pre-render every content page.

Edit: I already use generate but each click load the full page so my whole page is becoming white and the loading indicator cannot be displayed. What I wan to achieve is for instance keeping my `app.vue` loaded in which I have my Header and Footer and only load/reload `NuxtPage`:

```html
<template>
    <NuxtLoadingIndicator />
    <AppHeader></AppHeader>
    <NuxtPage />
    <AppFooter></AppFooter>
</template>
```

Is there any way to achieve that?

**Comment by xjccc** [comment]
If you want a loading for pages, you could see some disscussion on this https://github.com/nuxt/nuxt/issues/21721

or you could use `pnpm generate` generate static pages.

**Reply by AlexMili** [reply]
Thanks I will definitely check that.

I already use pre-rendered pages so static pages or am I missing something ?

**Reply by xjccc** [reply]
I don't know the scene of you. I think if you use `generate`, its doesn't has this problem..

I think its appear by vue spa mode.

**Reply by AlexMili** [reply]
I already use `generate` but each click load the full page so my whole page is becoming white and the loading indicator cannot be displayed. What I wan to achieve is for instance keeping my `app.vue` loaded in which I have my Header and Footer and only load/reload `NuxtPage`:
```html
<template>
    <NuxtLoadingIndicator />
    <AppHeader></AppHeader>
        <NuxtPage />
    <AppFooter></AppFooter>
</template>
```

**Reply by xjccc** [reply]
Could you use `render:html` hook, change your head\foorter in body? And add a loading for this?

```
<body>
<header></header>
<div id="loading"></div>
<div id="__nuxt"></div>
<footer></footer>
</body>

```

**Reply by AlexMili** [reply]
Well I guess but sounds tricky to do something I thought "easy". And you are talking about vue's hook or nuxt's hook for `render:html`?

**Reply by xjccc** [reply]
This [reply](https://github.com/nuxt/nuxt/issues/21721#issuecomment-2196559453)

**Comment by AlexMili** [comment, marked as answer]
I ended up using a state and `page:start` and `page:finish` to handle the loading:
```js
    const nuxtApp = useNuxtApp();
    const loading = useState("loading", () => true);
    nuxtApp.hook("page:start", () => {
        loading.value = true;
    });
    nuxtApp.hook("page:finish", () => {
        loading.value = false;
    });
```
